kani-driver/src/call_single_file.rs

```diff
@@ -160,6 +166,17 @@ impl KaniSession {
             flags.extend(str.split(' ').map(OsString::from));
         }
 
+        // This argument will select the Kani flavour of the compiler. It will be removed before
+        // rustc driver is invoked.
+        flags.push("--kani-compiler".into());
+
         flags
     }
 }
+
+/// This function can be used to convert a Kani compiler specific argument into a rustc one.
+/// We currently pass Kani specific arguments using the `--llvm-args` structure which is the
+/// hacky mechanism used by other rustc backend to receive arguments unknown to rustc.
```

Comment:
Can you elaborate on how llvm-args affects the cache?

Reply:
Sure